Rest API dependency version bump

zoliMolnar opened the pull request.

Commit:
[rest] Upgrade spring version, update maven dependencies in pom.xml...

... for snowowl.api.rest
- Remove old swagger ui use springfox's swagger 2 instead
- Remove old configuration xmls application-context.xml,
security-context.xml, servlet-context.xml and move everything to
ServicesConfiguration.class, SecurityConfiguration.class
- Move maven depenecies from snowowl-parent to the target platform
dependencies to avoid unnecessary additions of jars
- Remove predefined jars from snowowl.api.rest
- Modify web.xml
- Modify rest services to use the new spring annotations

Commit:
[rest] Upgrade spring version, update maven dependencies in pom.xml...

... for snowowl.snomed.api.rest
- Add js files to the static folder
- Add .gitignore under WEB-INF/lib
- Remove old swagger ui use springfox's swagger 2 instead
- Remove old configuration xmls application-context.xml,
security-context.xml, servlet-context.xml and move everything to
ServicesConfiguration.class, SecurityConfiguration.class
- Move maven depenecies from snowowl-parent to the target platform
dependencies to avoid unnecessary additions of jars
- Remove predefined jars from snowowl.api.rest
- Modify web.xml
- Modify rest services to use the new spring annotations

Commit:
[rest] Rename config files to match which packages they belong to...

- Convert api-description.html to markdown
- Change @Controller annotations to @RestController ones
- Remove unneeded jars from .classpath files
- Remove unneded build dependencies from build.properties, MANIFEST.MF
- Add missing favicon.png to the static folder

Commit:
[rest] Rename controller classes to match package wise naming convention

... and change annotaions for proper swagger ui representation

apeteri commented on core/com.b2international.snowowl.api.rest/src/com/b2international/snowowl/api/rest/SnowowlApiConfig.java:
```diff
@@ -48,108 +63,93 @@
+@Configuration
+@ComponentScan
+@Import({ SnowowlSecurityConfig.class })
+@PropertySource(value="classpath:com/b2international/snowowl/api/rest/service_configuration.properties")
```
Do the &commat;`Value` annotations work in the code below (and in other classes which use it), are you seeing the expected values from the property file in Swagger's API info section? If not, this snippet is also needed:

```
@Bean
public static PropertySourcesPlaceholderConfigurer ppc() throws IOException {
    return new PropertySourcesPlaceholderConfigurer();
}
```

zoliMolnar replied:
The @Value annotations work as intended

apeteri replied:
(A random GitHub user just got mentioned; talking about Java annotations in comments is hard. Use `&commat;` to escape the &commat; symbol.)

zoliMolnar replied:
I'll keep that in mind for future references, thank you.

apeteri commented on snomed/com.b2international.snowowl.snomed.api.rest/src/com/b2international/snowowl/snomed/api/rest/config/SnomedSecurityConfig.java:
```diff
@@ -15,27 +15,35 @@
  * @since 7.0
  */
 @Configuration
-@EnableGlobalMethodSecurity(securedEnabled=true, proxyTargetClass=true)
```
Note: This annotation corresponds to this bit in the .xml version:  `<global-method-security secured-annotations="enabled" proxy-target-class="true" />`, the corresponding feature is documented here: http://www.baeldung.com/spring-security-method-security

It got removed there, and it got removed here; I don't remember using &commat;`Secured` annotations on any method at the moment, so this _might_  be OK. Otherwise put it back in! 😄

zoliMolnar replied:
There are no methods which use @secured annotations so I've removed them, should I still add them back in any case?

apeteri commented on snomed/com.b2international.snowowl.snomed.api.rest/src/com/b2international/snowowl/snomed/api/rest/config/SnomedSecurityConfig.java:
```diff
@@ -57,17 +65,7 @@ protected void configure(final HttpSecurity http) throws Exception {
 
-	@Bean
-	@Override
-	public AuthenticationManager authenticationManagerBean() throws Exception {
```
This method should be kept! As I understand it, the flow of execution goes something like this:

- This configuration class inherits from `WebSecurityConfigurerAdapter`, which allows it to mess around with the `AuthenticationManagerBuilder` in the method above, before the manager is built;
- We set the provider on the builder to `authenticationProvider`;
- We should also expose the built `AuthenticationManager`, but if this dummy override is not present, it will not be available.

The reason is that the superclass does not have a &commat;`Bean` annotation on this method. See `WebSecurityConfigurerAdapter#authenticationManagerBean()`'s javadoc.

Commit:
[rest] Remove unneeded api description from properties and config add...

... back authenticationManagerBean override to
SnomedSecurityConfig.class

Commit:
[rest] Fix typo in service_configuration.properties add back ...

... authenticationManagerBean override to SnowowlSecurityConfig.class,
correctly use apiDescription field in ApiInfo constructor instead of
hardcoded value

Commit:
[rest] Remove IAuthenticationService and AuthenticationServiceImpl as...

... it's functionaly was inlined to SnowOwlAuthenticationProvider.class
Remove dependecies in manifests and remove bean from
SnomedApiConfig.class

zoliMolnar commented:
Thank you!
